src/popup/components/BackToTop.vue

```diff
@@ -1,78 +1,143 @@
+        v-if="isVisible"
+        class="back-to-top-btn-container"
+      >
+        <BtnPlain
+          class="back-to-top-btn"
+          @click="scrollTop"
+        >
+          <Chevron class="chevron" />
+        </BtnPlain>
+      </div>
+    </transition>
+  </div>
 </template>
 
-<script>
+<script lang="ts">
+import {
+  onBeforeUnmount,
+  onMounted,
+  defineComponent,
+  ref,
+} from 'vue';
 import BtnPlain from './buttons/BtnPlain.vue';
 import Chevron from '../../icons/chevron.svg?vue-component';
 
-export default {
+export default defineComponent({
   components: {
     BtnPlain,
     Chevron,
   },
-  data() {
-    return {
-      isVisible: false,
-    };
+  props: {
+    sticky: { type: Boolean, default: false },
   },
-  mounted() {
-    if (this.$el?.parentNode) {
-      this.$el.parentNode.addEventListener('scroll', this.handleVisibility);
-    }
-  },
-  beforeUnmount() {
-    if (this.$el?.parentNode) {
-      this.$el.parentNode.removeEventListener('scroll', this.handleVisibility);
+  setup() {
+    const el = ref<HTMLElement>();
+    const isVisible = ref(false);
+
+    function handleVisibility() {
+      if (!el.value?.parentNode) return;
+      isVisible.value = (el.value?.parentNode as HTMLElement)?.scrollTop > 800;
     }
-  },
-  methods: {
-    handleVisibility() {
-      const parent = this.$el?.parentNode;
-      if (!parent) return;
-      this.isVisible = parent.scrollTop > 0;
-    },
-    scrollTop() {
-      const parent = this.$el.parentNode;
-      parent.scrollTo({
+
+    function scrollTop() {
+      (el.value?.parentNode as HTMLElement)?.scrollTo({
         top: 0,
         behavior: 'smooth',
       });
-    },
+    }
+
+    onMounted(() => {
+      if (el.value?.parentNode) {
+        el.value?.parentNode.addEventListener('scroll', handleVisibility);
+      }
+    });
+
+    onBeforeUnmount(() => {
+      if (el.value?.parentNode) {
+        el.value?.parentNode.removeEventListener('scroll', handleVisibility);
+      }
+    });
+
+    return {
+      el,
+      isVisible,
+      scrollTop,
+    };
   },
-};
+});
 </script>
 
 <style lang="scss" scoped>
 @use '../../styles/variables';
 @use '../../styles/mixins';
 
 .back-to-top {
-  @include mixins.flex(center, center);
+  .back-to-top-spacer {
+    width: 100%;
+    height: 40px;
+  }
+
+  .back-to-top-btn-container {
+    @include mixins.flex(flex-end, center);
 
-  width: 100%;
-  height: 40px;
-  background: rgba(variables.$color-white, 0.1);
+    pointer-events: none;
+    position: fixed;
+    z-index: variables.$z-index-header;
+    bottom: 0;
+    left: 0;
+    width: 100%;
+    padding: 12px 4px 4px 0;
+    background: variables.$color-bg-4;
+    background:
+      linear-gradient(
+        180deg,
+        rgba(variables.$color-bg-4, 0) 5%,
+        rgba(variables.$color-bg-4, 0.8) 60%,
+        rgba(variables.$color-bg-4, 0.9) 100%
+      );
 
-  &:hover {
-    background: rgba(variables.$color-white, 0.08);
+    .back-to-top-btn {
```

Comment:
```suggestion
    &-btn {
```

Reply:
That would produce the following class name: `back-to-top-btn-container-btn`